src/template/transaction/builder.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { z } from 'zod';
+import { JSONBigInt } from '../../utils/bigint';
+import {
+  TxTemplateInstruction,
+  TransactionTemplate,
+  RawInputInstruction,
+  UtxoSelectInstruction,
+  AuthoritySelectInstruction,
+  RawOutputInstruction,
+  DataOutputInstruction,
+  TokenOutputInstruction,
+  AuthorityOutputInstruction,
+  ShuffleInstruction,
+  ChangeInstruction,
+  CompleteTxInstruction,
+  ConfigInstruction,
+  SetVarInstruction,
+} from './instructions';
+
+// Helper schemas to validate the arguments of each command in the builder args
+const RawInputInsArgs = RawInputInstruction.omit({ type: true });
+const UtxoSelectInsArgs = UtxoSelectInstruction.omit({ type: true });
+const AuthoritySelectInsArgs = AuthoritySelectInstruction.omit({ type: true });
+const RawOutputInsArgs = RawOutputInstruction.omit({ type: true });
+const DataOutputInsArgs = DataOutputInstruction.omit({ type: true });
+const TokenOutputInsArgs = TokenOutputInstruction.omit({ type: true });
+const AuthorityOutputInsArgs = AuthorityOutputInstruction.omit({ type: true });
+const ShuffleInsArgs = ShuffleInstruction.omit({ type: true });
+const ChangeInsArgs = ChangeInstruction.omit({ type: true });
+const CompleteTxInsArgs = CompleteTxInstruction.omit({ type: true });
+const ConfigInsArgs = ConfigInstruction.omit({ type: true });
+const SetVarInsArgs = SetVarInstruction.omit({ type: true });
+
+export class TransactionTemplateBuilder {
+  template: z.infer<typeof TransactionTemplate>;
+
+  constructor() {
+    this.template = [];
+  }
+
+  static from(instructions: z.input<typeof TransactionTemplate>): TransactionTemplateBuilder {
```

Comment:
Obs: `z.input<T>` means that this accepts the "unparsed" instruction.
When the type is `z.infer<T>` it means the output of the parsed data.